### PR TITLE
Feature/fbf accounts

### PR DESCRIPTION
**Pull request by steffansluis**
This PR adds `feedbackfruits-ui-accounts`, as well as the appropriate routes to handle registering, logging in and out and linked external accounts. At the moment, only the `/mygoals` route requires being logged in, since logging in doesn't work perfectly yet due to the accounts service not anticipating different clients than the FeedbackFruits platform. This is outside of the scope of this PR and will be fixed in the near future.

Furthermore, the `login.hbs` template renders the different options for logging in, which at the moment are GitHub and email/password. However, for the same reasons as mentioned above, this doesn't work yet since the identity providers are scoped under the client, which for the moment is still 'FeedbackFruits', which doesn't offer authentication with GitHub (yet).

To summarize, this PR adds all the necessary building blocks.

### CODE REVIEW

**Comment by steffansluis** [comment]
@WouterKvG @bartist Please review!

File: app/routes/index/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Ember from 'ember';
 
 export default Ember.Route.extend({
-  beforeModel() {
+  redirect() {
     this.transitionTo('index.mygoals');
   }
 });
```
Comment (bartist) [inline app/routes/index/index.js]: Why was this replaced from "/index.js" to "index/index.js"?
I understand the change to redirect instead of beforeModel(), but the replacement prevents the redirection

**Comment by bartist** [comment]
when serving the site, I got this in my log:
DEBUG: -------------------------------
ember.debug.js:6440 DEBUG: Ember             : 2.5.1
ember.debug.js:6440 DEBUG: Ember Data        : 2.5.2
ember.debug.js:6440 DEBUG: jQuery            : 2.2.3
ember.debug.js:6440 DEBUG: Ember Simple Auth : 1.1.0
ember.debug.js:6440 DEBUG: -------------------------------
http://localhost.feedbackfruits.com:3001/identity_providers Failed to load resource: net::ERR_CONNECTION_REFUSED
ember.debug.js:28535 Error while processing route: login.index Ember Data Request GET http://localhost.feedbackfruits.com:3001/identity_providers returned a 0
Payload (Empty Content-Type)
 Error: Ember Data Request GET http://localhost.feedbackfruits.com:3001/identity_providers returned a 0
Payload (Empty Content-Type)

    at new Error (native)
    at Error.EmberError (http://localhost:4200/assets/vendor.js:26691:21)
    at Error.AdapterError (http://localhost:4200/assets/vendor.js:76431:16)
    at handleResponse (http://localhost:4200/assets/vendor.js:87885:14)
    at hash.error (http://localhost:4200/assets/vendor.js:87966:29)
    at fire (http://localhost:4200/assets/vendor.js:3454:31)
    at Object.self.fireWith [as rejectWith] (http://localhost:4200/assets/vendor.js:3584:7)
    at done (http://localhost:4200/assets/vendor.js:9054:14)
    at XMLHttpRequest.<anonymous> (http://localhost:4200/assets/vendor.js:9410:10)logError @ ember.debug.js:28535
ember.debug.js:32096 Error: Ember Data Request GET http://localhost.feedbackfruits.com:3001/identity_providers returned a 0
Payload (Empty Content-Type)

    at new Error (native)
    at Error.EmberError (http://localhost:4200/assets/vendor.js:26691:21)
    at Error.AdapterError (http://localhost:4200/assets/vendor.js:76431:16)
    at handleResponse (http://localhost:4200/assets/vendor.js:87885:14)
    at hash.error (http://localhost:4200/assets/vendor.js:87966:29)
    at fire (http://localhost:4200/assets/vendor.js:3454:31)
    at Object.self.fireWith [as rejectWith] (http://localhost:4200/assets/vendor.js:3584:7)
    at done (http://localhost:4200/assets/vendor.js:9054:14)
    at XMLHttpRequest.<anonymous> (http://localhost:4200/assets/vendor.js:9410:10)

Seems something on my machine or in the code does not do what it is suppost to do

**Comment by bartist** [comment]
Code-wise: it looks good!
If the issue above can be fixed one way or another, that would be great.
If it is an issue on my machine or it is fixed, feel free to merge

File: app/routes/index/index.js (hunk 1 above)
Comment (steffansluis) [inline app/routes/index/index.js]: Since we have an `index` directory in routes, I though it was nicer to put it there rather than have an `index.js` file and an `index` directory, but that is pretty much the only reason. I'm not sure if there is a functional difference, but it appears to work the same way.

**Comment by steffansluis** [comment]
@bartist The issue is because you don't have the acocunts server running locally. You could set it up or run with `EMBER_ENV=staging ember serve`. The issues mentioned in the PR description will still apply though.
